openassessment/xblock/ui_mixins/mfe/page_context_serializer.py

```diff
@@ -237,58 +299,46 @@ class PageDataSerializer(Serializer):
     Requires context to differentiate between Assessment and Submission views
     """
 
-    require_context = True
+    requires_context = True
 
     progress = ProgressSerializer(source="*")
-    submission = SerializerMethodField()
-    rubric = RubricConfigSerializer(source="*")
+    response = SerializerMethodField()
     assessment = SerializerMethodField()
 
     def to_representation(self, instance):
-        # Loading workflow status causes a workflow refresh
-        # ... limit this to one refresh per page call
-        if not self.context.get("step"):
-            active_step = instance.workflow_data.status or "submission"
-            self.context.update({"step": active_step})
+        if "step" not in self.context:
+            raise ValidationError("Missing required context: step")
+        if "view" not in self.context:
+            raise ValidationError("Missing required context: view")
 
         return super().to_representation(instance)
 
-    def _can_jump_to_step(self, workflow_step, workflow_data, step_name):
```

Comment:
This is good, we should pull as much of this logic out of the serializer as we can